src/rdrand.c

```diff
@@ -32,23 +32,16 @@
 #include "tinycthread.h"
 
 int rd_jitter(int low, int high) {
-        int rand_num;
-#if HAVE_RAND_R
```

Comment:
This makes it so that rand_r() is required on non-windows platforms, which we can't really assert.

I think a better option is to use HAVE_RAND_R and only provide the musl-inspired rd_rand_r() if HAVE_RAND_R is false.